sourcegraph/sourcegraph
# extensibility: implement featured extensions

- Add `featuredExtensions` field to `ExtensionRegistry` GraphQL type
- Add `registry/extensions/featured` endpoint to HTTP API to allow other instances to ask sourcegraph.com for featured extensions (allows us to update featured extensions independently of releases)
- Implement featured extensions section + card type ([Figma](https://www.figma.com/file/hciWKNbllHeHWoYcpfGQp2/RFC-305-Next-Up-Extensions-Registry-%22Marketplace%22-Concepts-%5BReview%5D?node-id=714%3A0), closes #21009, closes #20780)

![Screenshot from 2021-06-03 17-52-47](https://user-images.githubusercontent.com/37420160/120716894-ac4f1580-c494-11eb-8582-cc711b103923.png)

- Should we only display a featured extension once, not again in its primary category's section? 

### Testing locally

1. run `SOURCEGRAPHDOTCOM_MODE=1 enterprise/dev/start.sh` in SG repo root
1. create a few test extensions (or clone some of our extensions) and publish them to your local registry with the [src cli](https://docs.sourcegraph.com/extensions/authoring/publishing)
1. add those extension IDs to [this](https://github.com/sourcegraph/sourcegraph/pull/21665#discussion_r645675033) slice and wait for frontend to recompile
1. reload the registry page and see the featured extensions section in action (only displayed when no category is selected and search query is empty,
